src/tests/test_collection.py

```diff
@@ -577,6 +577,55 @@ def test_filters_in(client: vecs.Client) -> None:
         )
 
 
+def test_filters_contains(client: vecs.Client) -> None:
+    bar = client.get_or_create_collection(name="bar", dimension=4)
+
+    records = [
+        ("0", [0, 0, 0, 0], {"a": 1, "b": 2}),
+        ("1", [1, 0, 0, 0], {"a": [1, 2, 3]}),
+        ("2", [1, 1, 0, 0], {"a": {"1": "2"}}),
+        ("3", [0, 0, 0, 0], {"a": "1"}),
+    ]
+
+    bar.upsert(records)
+    bar.create_index()
+
+    # Test $contains operator for int value
+    assert bar.query(
+        data=[0, 0, 0, 0],
+        limit=3,
+        filters={"a": {"$contains": 1}},
+    ) == ["0", "1"]
```

Comment:
Should `"0"` be in this result? i.e. when the value of `a` is a scalar that equals the `contains` clause, should that count as a match?

did you expect this behavior or did you add `"0"` because that's what it did by default?

If we wanted to exclude `"0"` I think we could add a clause to say that `jsonb_typeof((table.metadata -> 'a')) = 'array'

Reply:
My first thought is we might need to change some of these behaviors.
If we're trying to match python behavior:

```py
1 in [1] # True
{"a": 1} in {"a": 1, "b": 2} # TypeError
1 in 1 # TypeError
```